runtime/syntax/javascript.vim

```diff
@@ -40,7 +40,7 @@ syn region  javaScriptStringT	       start=+`+  skip=+\\\\\|\\`+  end=+`+	contai
 syn region  javaScriptEmbed	       start=+${+  end=+}+	contains=@javaScriptEmbededExpr
 
 syn match   javaScriptSpecialCharacter "'\\.'"
-syn match   javaScriptNumber	       "-\=\<\d\+L\=\>\|0[xX][0-9a-fA-F]\+\>"
+syn match   javaScriptNumber	       "-\=\<\d\+L\=\>\|0[xX][0-9a-fA-F_]\+\>"
```

Comment:
Thanks.
CC'ing the maintainer @fleiner since he's on github.

If I understand well, the underscore for digit separator is only a proposal for Javascript. See:
https://github.com/tc39/proposal-numeric-separator
I don't think it is standard yet. However, it is already implemented in at least Chrome, Firefox and Safari, so I think it's OK to support it in Vim.

I see that the support is still incomplete with your patch though:

- underscore should also be allowed in decimal numbers (e.g. `123_456`) yet they are not recognized as javaScriptNumber even after this patch
- binary and octal numbers are not recognized at all as javaScriptNumber even without underscores. Example of valid javaScript numbers not yet recognized: `0b101`, `0b1010_1010,` `0o1234`, `0o12_34`)
- the underscore should only be allowed between digits. Vim should not recognize the following numbers as javaScriptNumber: `123_`, `0x12af_`, `0x_12af`, `3._1415F` …

Slightly related, I see that underscore separators are also not fully supported for the java syntax (at least for hex and binary numbers).  Java and JavaScript syntax files have the same maintainer.  For Java, see: https://docs.oracle.com/javase/7/docs/technotes/guides/language/underscores-literals.html

I will check whether there are other languages for which digit separators are not supported. This page https://rosettacode.org/wiki/Numeric_separator_syntax gives several languages where digit separators should be supported.